packages/hash/frontend/src/components/hooks/blockProtocolFunctions/useBlockProtocolDeleteLinkedAggregations.ts

```diff
@@ -0,0 +1,67 @@
+import { useMutation } from "@apollo/client";
+
+import { BlockProtocolDeleteLinkedAggregationsFunction } from "blockprotocol";
+import { deleteLinkedAggregationMutation } from "@hashintel/hash-shared/queries/link.queries";
+import { useCallback } from "react";
+import {
+  DeleteLinkedAggregationMutation,
+  DeleteLinkedAggregationMutationVariables,
+} from "../../../graphql/apiTypes.gen";
+
+export const useBlockProtocolDeleteLinkedAggregations = (): {
+  deleteLinkedAggregations: BlockProtocolDeleteLinkedAggregationsFunction;
+  deleteLinkedAggregationsLoading: boolean;
+  deleteLinkedAggregationsError: any;
+} => {
+  const [
+    runDeleteLinkedAggregationsMutation,
+    {
+      loading: deleteLinkedAggregationsLoading,
+      error: deleteLinkedAggregationsError,
+    },
+  ] = useMutation<
+    DeleteLinkedAggregationMutation,
+    DeleteLinkedAggregationMutationVariables
+  >(deleteLinkedAggregationMutation);
+
+  const deleteLinkedAggregations: BlockProtocolDeleteLinkedAggregationsFunction =
+    useCallback(
+      async (actions) => {
+        const results: boolean[] = [];
+        // TODO: Support multiple actions in one GraphQL mutation for transaction integrity and better status reporting
+        for (const action of actions) {
+          if (!action.sourceAccountId) {
+            throw new Error("deleteLinks needs to be passed a sourceAccountId");
+          }
+
+          if (!action.sourceEntityId) {
+            throw new Error("deleteLinks needs to be passed a sourceEntityId");
+          }
```

Comment:
I've made a comment elsewhere – I think we can remove the requirement to pass a `sourceEntityId`. It is not required to identify the aggregation, and not required for sharding.

Reply:
Addressed in https://github.com/hashintel/hash/pull/497/commits/413855ddf7ea2991b42e473c18f55e0fa9df2318